server/core/src/main/java/org/infinispan/server/core/backup/BackupManagerImpl.java

```diff
@@ -45,6 +45,7 @@
    final Lock restoreLock;
    final Map<String, DefaultCacheManager> cacheManagers;
    final Map<String, BackupRequest> backupMap;
+   final Map<String, CompletionStage<Void>> restoreMap;
```

Comment:
Currently a `ClusteredLock` is used if a transport is configured to ensure that only a single backup or restore can occur at anyone time.

The maps are local, because the resources that they create/read are also local to the server. For the operator, loadbalancing is not an issue, as the the backup/restore operation is always executed on a single kubernetes pod and the controller interacts with this pod directly.

I agree that we can improve this situation though. We could utilise an internal cache to track the progress of a backup/restore state, including the node it's executing on. This would allow us to handle status requests from all Infinispan nodes, as well as allowing a `3xx` response to be sent when an attempt is made to retrieve a backup on a node that does not contain the files.

Reply:
> a 3xx response to be sent when an attempt is made to retrieve a backup on a node that does not contain the files.

The redirection has issues by itself. IMO, the REST APIs should be consistent if part of the ```cluster``` resource, otherwise we run various issues such as CORS (can't call an internal server unless explicitly allowing in config), network (can't reach an internal load balanced server) and consistency (same URL returns 2xx or 3xx when using load balancer).

One way to solve it is to have every REST server accept the request and forward it to the internal node if the request can only be satisfied by a local node, but I think all these matters should be handled in a separate JIRA since it would make this PR a lot bigger

Reply:
Will you create a JIRA @ryanemerson or should I?

Reply:
I've created [ISPN-12393](https://issues.redhat.com/browse/ISPN-12393)